packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -562,31 +603,41 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
         break;
     }
 
+    final Size viewport = widget.renderObject.size;
+    point = Offset(
```

Comment:
nit: flutter style would be to add a trailing comma and indent by just two spaces:

```dart
point = Offset(
  point.dx.clamp(0.0, viewport.width),
  point.dy.clamp(0.0, viewport.height),
);
```

Reply:
Done.